bgpd/bgp_ecommunity.c

```diff
@@ -619,11 +784,24 @@ static int ecommunity_rt_soo_str(char *buf, size_t bufsz, const uint8_t *pnt,
 
 		len = snprintf(buf, bufsz, "%s%u:%u", prefix, eas.as, eas.val);
 	} else if (type == ECOMMUNITY_ENCODE_AS) {
-		eas.as = (*pnt++ << 8);
-		eas.as |= (*pnt++);
-		pnt = ptr_get_be32(pnt, &eas.val);
+		if (ecom_size == ECOMMUNITY_SIZE) {
+			eas.as = (*pnt++ << 8);
```

Comment:
I think it would be more readable using `ECOMMUNITY_SIZE` instead of numbers everywhere. Same with 16.

Reply:
added a comment to explain that.
8 is not ecommunity size.